test/test_iam_assume_token_manager.py

```diff
@@ -0,0 +1,180 @@
+# coding: utf-8
+
+# Copyright 2019, 2024 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=missing-docstring
+import json
+import logging
+import time
+
+import jwt
+import responses
+
+from ibm_cloud_sdk_core import IAMAssumeTokenManager
+from .utils.logger_utils import setup_test_logger
+
+
+MY_PROFILE_ID = 'my-profile-id'
+MY_PROFILE_CRN = 'my-profile-crn'
+MY_PROFILE_NAME = 'my-profile-name'
+MY_ACCOUNT_ID = 'my-account_id'
+
+
+setup_test_logger(logging.ERROR)
+
+
+def _get_current_time() -> int:
+    return int(time.time())
+
+
+ACCESS_TOKEN_LAYOUT = {
+    "username": "dummy",
+    "role": "Admin",
+    "permissions": ["administrator", "manage_catalog"],
+    "sub": "admin",
+    "iss": "sss",
+    "aud": "sss",
+    "uid": "sss",
+    "iat": _get_current_time(),
+    "exp": _get_current_time() + 3600,
+}
+
+ACCESS_TOKEN = jwt.encode(
+    ACCESS_TOKEN_LAYOUT, 'secret', algorithm='HS256', headers={'kid': '230498151c214b788dd97f22b85410a5'}
+)
+
+
+@responses.activate
+def test_request_token_with_profile_id():
+    iam_url = "https://iam.cloud.ibm.com/identity/token"
+    response = {
+        "access_token": ACCESS_TOKEN,
+        "token_type": "Bearer",
+        "expires_in": 3600,
+        "expiration": _get_current_time() + 3600,
+        "refresh_token": "jy4gl91BQ",
+    }
+    responses.add(responses.POST, url=iam_url, body=json.dumps(response), status=200)
```

Comment:
No. I mean it's not necessary just for checking the outgoing requests (which the test does) because the mock response is generic enough to cover both HTTP calls and it's not a stack, so matched mock responses are not removed. That said, I changed the reworked these token manager tests a little bit, so now it should mimic a real world scenario, because it returns different results based on the `grant_type`. I also added a test for the `get_token()` method where the returned access token is checked, just to make sure the final result contains the correct one. So please take a look at the new tests, just to make sure I didn't miss anything.:)